hw12/src/CacheControlFilter.java

```diff
@@ -0,0 +1,53 @@
+import java.util.Arrays;
+
+public class CacheControlFilter implements Filter{
+    public static final String[] VALID_OPTIONS = {
+        "max-age",
+        "max-stale",
+        "min-fresh"};
+    public static final String[] VALID_FLAGS = {
+        "no-cache",
+        "no-store",
+        "no-transform",
+        "only-if-cached",
+        "stale-if-error"};
+
+    public static final String CACHE_CONTROL_PREFIX = "Cache-Control:";
+
+    @Override
+    public FilterResult doFilter(Request request) {
+        FilterResult result = new FilterResult();
+        result.result = true;
+        for (String header : request.getHeaders()){
+            if (header.startsWith(CACHE_CONTROL_PREFIX)){
```

Comment:
Hign "if" complexity. You can simplify your code in the following way:
instead of 
for{
  if(condition){
    if logic;
  } else {
    else logic
  }
}
use:
for{
  if(!condition){
        else logic
    break;
  } 
  if logic;
}
 in such case your code will be more readable and you will have less embedded code scopes brackets.